analyzers/tests/SonarAnalyzer.Test/Rules/AspNet/ControllersHaveTooManyResponsibilitiesTest.cs

```diff
@@ -0,0 +1,51 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis.CSharp;
+using SonarAnalyzer.Rules.CSharp;
+
+namespace SonarAnalyzer.Test.Rules;
+
+[TestClass]
+public class ControllersHaveTooManyResponsibilitiesTest
+{
+    private readonly VerifierBuilder builder =
+        new VerifierBuilder<ControllersHaveTooManyResponsibilities>();
+
+#if NET
+    private static IEnumerable<MetadataReference> AspNetCoreReferences =>
+    [
+        AspNetCoreMetadataReference.MicrosoftAspNetCoreMvcAbstractions,
+        AspNetCoreMetadataReference.MicrosoftAspNetCoreMvcCore,
+        AspNetCoreMetadataReference.MicrosoftAspNetCoreMvcViewFeatures,
+    ];
+
+    [TestMethod]
+    public void ControllersHaveTooManyResponsibilities_CS() =>
+        builder
+            .AddReferences(AspNetCoreReferences)
+            .AddReferences([CoreMetadataReference.SystemComponentModel]) // For IServiceProvider
+            .AddReferences(NuGetMetadataReference.MicrosoftExtensionsDependencyInjectionAbstractions("8.0.1")) // For IServiceProvider extensions
+            .WithBasePath("AspNet")
+            .AddPaths("ControllersHaveTooManyResponsibilities.CSharp12.cs")
+            .WithLanguageVersion(LanguageVersion.CSharp12)
```

Comment:
Until the circle is finalized, the outcome of the circle should not be applied to production work.
Let's do such tests in another context, possibly in a branch that doesn't get merged.

Reply:
I would agree with this in general, but we do not really have a hard-enforced convention about how to structure our UTs.
And I do not see any danger with doing this, at least not until C#13 is released.
I will leave it up to you, since you make a valid argument.